### PR TITLE
Add generic level 1b and 2 ascat data readers

### PR DESCRIPTION
**Pull request by mazweg**
data reader classes for lvl1 and lvl2 to handle all formats from the EUMETSAT Data Center (bufr, native_eps, netcdf, hdf5).  

Travis tests failing because of missing szf files (too big for github).

### CODE REVIEW

File: ascat/read_native/nc.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+            start date
+        end_date : datetime.date or datetime.datetime
+            end date
+
+        Returns
+        -------
+        dates : list
+            list of datetimes
+        """
+        file_list = []
```
Comment (cpaulik) [inline ascat/read_native/nc.py]: I might be wrong but this part seems to be repeated often. Could we make a Baseclass or a Mixin that implements this if it is really identical?

**Comment by cpaulik** [comment]
How big are the szf files?

**Comment by mazweg** [comment]
approx. 170MB - we will host a git repository with the test data on our own webserver soon

File: ascat/read_native/nc.py (hunk 1 above)
Comment (sebhahn) [reply ascat/read_native/nc.py]: I have reviewed the whole code yet, but I agree that a base class makes sense if there is a lot of overlap. @mazweg could you look into that please

**Comment by sebhahn** [comment]
There are two problems w.r.t. to file size: (i) Github allows only files up to 100 MB and (ii) the bandwidth is only 1 GB per month, otherwise you have to pay. Our solution is now hosting the test data in a git repository on TUW GEO webserver. As soon as the new webserver is running we try to set this up correctly.

File: ascat/read_native/nc.py (hunk 1 above)
Comment (mazweg) [reply ascat/read_native/nc.py]: yes I am looking at it atm and it looks like a base class would indeed make sense.

**Comment by cpaulik** [comment]
Gitlab.com might be an option for hosting the testdata. The limit there is 10GB with unlimited bandwith. Might be better for open source than a TUW GEO server that has to be maintained etc.

**Comment by sebhahn** [comment]
We will try if and how it works, because if gitlab.com is changing their limits at some point we have the same problem

**Comment by sebhahn** [comment]
@mazweg: Good work! :+1: here are some comments:

- [ ] Fix pep8 issues (e.g. line length only 80), add a pep8 check to your IDE
- [ ] Add copyright header to base.py
- [ ] Do not distinguish between Metop-A and Metop-B, maybe use a placeholder '?' that the same class can read Metop-A and Metop-B
- [ ] Check unused imports in level1.py and level2.py (`timedelta`, `MultitemporalImageBase`)
- [ ] Don't use such large ifs like in `eps2generic`, please create two functions for if and else clause
- [ ] Docstrings for last functions in eps_native.py and hdf5.py missing
- [ ] Add examples to the documention how to read a L1b and L2 file + simple plot of swath

**Pull request merged**